server/app/mutations/grid_services/update.rb

```diff
@@ -75,28 +75,34 @@ def execute
       attributes[:health_check] = self.health_check if self.health_check
       attributes[:volumes_from] = self.volumes_from if self.volumes_from
 
+      embeds_changed = false
+
       if self.links
         attributes[:grid_service_links] = build_grid_service_links(
           self.grid_service.grid_service_links.to_a,
           self.grid_service.grid, grid_service.stack, self.links
         )
+        embeds_changed ||= attributes[:grid_service_links] != self.grid_service.grid_service_links.to_a
       end
 
       if self.hooks
         attributes[:hooks] = self.build_grid_service_hooks(self.grid_service.hooks.to_a)
+        embeds_changed ||= attributes[:hooks] != self.grid_service.hooks.to_a
```

Comment:
This might run into some issues with the `hooks` input param being an unordered hash of hook types, so the order of the resulting hooks might change... but ATM there is only a single type of hook (`post_start`), so this is moot.

The `build_grid_service_hooks` needs fixing to `sort` the `hooks` keys to guarantee stable ordering across updates though, if we ever do add more hook types.